src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php

```diff
@@ -136,4 +135,11 @@ public function handleAuthenticationFailure(AuthenticationException $authenticat
             is_object($response) ? get_class($response) : gettype($response)
         ));
     }
+
+    private function dispatchEvent($eventName, Event $event)
```

Comment:
I would not introduce this method (same below). We have this check inlined anywhere else too (take a look at the `AuthenticationProviderManager` for example).